File: Solutions/Easy/1351. Count Negative Numbers in a Sorted Matrix/Solution.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        result: int = 0
        row_pointer:int = len(grid[0]) - 1
        col_height: int = len(grid)
        for i, r in enumerate(grid):
            while(r[row_pointer] < 0):
                result += col_height - i
                row_pointer -= 1
                if(row_pointer == -1):
                    return result
        return result

File: Solutions/Easy/1351. Count Negative Numbers in a Sorted Matrix/Solution.cs
public class Solution {
    public int CountNegatives(int[][] grid) {
        int n = grid[0].Length, result = 0;
        foreach(int[] row in grid){
            result += BinarySearch(row, n);
        }
        return result;
    }

    private int BinarySearch(int[] row, int n){
        int left = 0, right = n - 1, res = -1;
        while(left <= right){
            int mid = left + (right - left) / 2;
            if (row[mid] >= 0){
                left = mid + 1;
            }
            else{
                right = mid - 1;
                res = mid;
            }
        }
        return res == -1 ? 0 : n - res;
    }
}